src/libraries/System.Linq/tests/CastTests.cs

```diff
@@ -235,5 +234,111 @@ public void ForcedToEnumeratorDoesntEnumerate()
             var en = iterator as IEnumerator<string>;
             Assert.False(en != null && en.MoveNext());
         }
+
+        [Fact]
+        public void TargetTypeIsSourceType_Nop()
+        {
+            object[] values = new string[] { "hello", "world" };
```

Comment:
Is it a no-op even if the source enumerable is not an ICollection?

Reply:
Yes, that's already the case:
https://github.com/dotnet/runtime/blob/973ceee20c4807c780f72e24a1859a2c695ddf39/src/libraries/System.Linq/src/System/Linq/Cast.cs#L38-L41